Vialer/AccountViewController.m

```diff
@@ -154,11 +155,18 @@ - (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath
 #pragma mark - Editnumber delegate
 
 - (void)numberHasChanged:(NSString *)newNumber {
-    //Update the userdefaults
-    [[NSUserDefaults standardUserDefaults] setObject:newNumber forKey:@"MobileNumber"];
-    //Update the tableView Cell
-    UITableViewCell *myNumberCell = [self.tableView cellForRowAtIndexPath:[NSIndexPath indexPathForRow:MY_NUMBER_ROW inSection:NUMBERS_SECTION]];
-    myNumberCell.detailTextLabel.text = newNumber;
+    [[VoIPGRIDRequestOperationManager sharedRequestOperationManager] pushMobileNumber:newNumber success:^{
```

Comment:
Check or newNumber != nil by doing the length check you also do in pushMobileNumber function.